plugins/out_s3/s3.c

```diff
@@ -1075,7 +1084,38 @@ static int upload_data(struct flb_s3 *ctx, struct s3_file *chunk,
         m_upload->upload_state = MULTIPART_UPLOAD_STATE_CREATED;
     }
 
-    ret = upload_part(ctx, m_upload, body, body_size);
+    void *payload_buf = NULL;
+    size_t payload_size = 0;
+    ctx->isCompressed = FLB_FALSE;
+    if (ctx->compression == FLB_AWS_COMPRESS_GZIP) {
+        /* Map payload */
+
+        ret = flb_gzip_compress((void *) body, body_size, &payload_buf, &payload_size);
```

Comment:
We use the AWS wrapper around the flb compression code: https://github.com/fluent/fluent-bit/blob/master/plugins/out_s3/s3.c#L1336

Reply:
Saw it. will change it here